File: v0/HuntKingdom v0/src/ProduitBundle/Entity/Produit.php
<?php

namespace ProduitBundle\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;

/**
 * Produit
 *
 * @ORM\Table(name="produit")
 * @ORM\Entity(repositoryClass="ProduitBundle\Repository\ProduitRepository")
 */
class Produit
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;
    /**
     * @var string
     *
     * @ORM\Column(name="nom", type="string", length=255)
     */
    private $nom;

    /**
     * @var int
     *
     * @ORM\Column(name="quantite", type="integer")
     */
    private $quantite;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=255)
     */
    private $description;

    /**
     * @var float
     *
     * @ORM\Column(name="prix", type="float")
     */
    private $prix;

    /**
     * @var int
     *
     * @ORM\Column(name="idFournisseur", type="integer", nullable=true)
     */
    private $idFournisseur;

    /**
     * @var int
     *
     * @ORM\Column(name="etatPromo", type="integer")
     */
    private $etatPromo;

    /**
     * @var int
     *
     * @ORM\Column(name="garantie", type="integer")
     */
    private $garantie;

    /**
     * @ORM\ManyToOne(targetEntity="Categorie")
     * @ORM\JoinColumn(name="categorie", referencedColumnName="id")
     */
    private $categorie;
    /**
     * @var string
     * @Assert\File(mimeTypes={ "image/jpeg", "image/png"})
     * @ORM\Column(name="image", type="string")
     */
    private $image;

    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set nom
     *
     * @param string $nom
     *
     * @return Produit
     */
    public function setNom($nom)
    {
        $this->nom = $nom;

        return $this;
    }

    /**
     * Get nom
     *
     * @return string
     */
    public function getNom()
    {
        return $this->nom;
    }

    /**
     * Set quantite
     *
     * @param integer $quantite
     *
     * @return Produit
     */
    public function setQuantite($quantite)
    {
        $this->quantite = $quantite;

        return $this;
    }

    /**
     * Get quantite
     *
     * @return integer
     */
    public function getQuantite()
    {
        return $this->quantite;
    }

    /**
     * Set description
     *
     * @param string $description
     *
     * @return Produit
     */
    public function setDescription($description)
    {
        $this->description = $description;

        return $this;
    }

    /**
     * Get description
     *
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * Set prix
     *
     * @param float $prix
     *
     * @return Produit
     */
    public function setPrix($prix)
    {
        $this->prix = $prix;

        return $this;
    }

    /**
     * Get prix
     *
     * @return float
     */
    public function getPrix()
    {
        return $this->prix;
    }

    /**
     * Set idFournisseur
     *
     * @param integer $idFournisseur
     *
     * @return Produit
     */
    public function setIdFournisseur($idFournisseur)
    {
        $this->idFournisseur = $idFournisseur;

        return $this;
    }

    /**
     * Get idFournisseur
     *
     * @return integer
     */
    public function getIdFournisseur()
    {
        return $this->idFournisseur;
    }

    /**
     * Set etatPromo
     *
     * @param integer $etatPromo
     *
     * @return Produit
     */
    public function setEtatPromo($etatPromo)
    {
        $this->etatPromo = $etatPromo;

        return $this;
    }

    /**
     * Get etatPromo
     *
     * @return integer
     */
    public function getEtatPromo()
    {
        return $this->etatPromo;
    }

    /**
     * Set garantie
     *
     * @param integer $garantie
     *
     * @return Produit
     */
    public function setGarantie($garantie)
    {
        $this->garantie = $garantie;

        return $this;
    }

    /**
     * Get garantie
     *
     * @return integer
     */
    public function getGarantie()
    {
        return $this->garantie;
    }

    /**
     * Set image
     *
     * @param string $image
     *
     * @return Produit
     */
    public function setImage($image)
    {
        $this->image = $image;

        return $this;
    }

    /**
     * Get image
     *
     * @return string
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * Set categorie
     *
     * @param \ProduitBundle\Entity\Categorie $categorie
     *
     * @return Produit
     */
    public function setCategorie(\ProduitBundle\Entity\Categorie $categorie = null)
    {
        $this->categorie = $categorie;

        return $this;
    }

    /**
     * Get categorie
     *
     * @return \ProduitBundle\Entity\Categorie
     */
    public function getCategorie()
    {
        return $this->categorie;
    }
}

File: SPRINT DESKTOP/INTEGRATION/HuntKingdom 0.5 All interfaces switch/src/Entities/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import javafx.scene.image.ImageView;

/**
 *
 * @author khalil
 */
public class Produit {
      
    private int id;
    private  Categorie categorie;
    private String nom;
    private int quantite;
    private float prix;
    private String description;
    private int idFournisseur;
    private int etatPromo;
    private int garantie;
    private int tvaId;
    private String image;
    private Promotion promotion;

    private ImageView im;

    public Produit(int id, String nom, int quantite, float prix, String description, int etatPromo, int garantie, String image, Promotion promotion) {
        this.id = id;
        this.nom = nom;
        this.quantite = quantite;
        this.prix = prix;
        this.description = description;
        this.etatPromo = etatPromo;
        this.garantie = garantie;
        this.image = image;
        this.promotion = promotion;
    }

    public ImageView getIm() {
        return im;
    }

    public void setIm(ImageView im) {
        this.im = im;
    }
    //for affichage
    private String nomCat;

    public String getNomCat() {
        return nomCat;
    }

    public Produit(int id, Categorie categorie, String nom, int quantite, float prix, String description, int garantie) {
        this.id = id;
        this.categorie = categorie;
        this.nom = nom;
        this.quantite = quantite;
        this.prix = prix;
        this.description = description;
        this.garantie = garantie;
    }

    public void setNomCat(String nomCat) {
        this.nomCat = nomCat;
    }
    public Promotion getPromotion() {
        return promotion;
    }

    public void setPromotion(Promotion promotion) {
        this.promotion = promotion;
    }
    public Produit() {
     
    }

    @Override
    public String toString() {
        return "Produit{" + "id=" + id  +
                ", nom=" + nom + ", quantite=" + quantite + ", prix=" + prix + 
                ", description=" + description + ", etatPromo=" + etatPromo + 
                ", garantie=" + garantie + ", image=" + image +'}' ;
        
    }

    public Produit(int id, Categorie categorie, String nom, int quantite, float prix, String description, int garantie, String image) {
        this.id = id;
        this.categorie = categorie;
        this.nom = nom;
        this.quantite = quantite;
        this.prix = prix;
        this.description = description;
        this.garantie = garantie;
        this.image = image;
    }

    public Produit(Categorie categorie, String nom, int quantite, float prix, String description, int garantie, String image) {
        this.categorie = categorie;
        this.nom = nom;
        this.quantite = quantite;
        this.prix = prix;
        this.description = description;
        this.garantie = garantie;
        this.image = image;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Categorie getCategorie() {
        return categorie;
    }

    public void setCategorie(Categorie categorie) {
        this.categorie = categorie;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getIdFournisseur() {
        return idFournisseur;
    }

    public void setIdFournisseur(int idFournisseur) {
        this.idFournisseur = idFournisseur;
    }

    public int getEtatPromo() {
        return etatPromo;
    }

    public void setEtatPromo(int etatPromo) {
        this.etatPromo = etatPromo;
    }

    public int getGarantie() {
        return garantie;
    }

    public void setGarantie(int garantie) {
        this.garantie = garantie;
    }

    public int getTvaId() {
        return tvaId;
    }

    public void setTvaId(int tvaId) {
        this.tvaId = tvaId;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Produit(String nom, float prix, String description, int garantie, String nomCat) {
        this.nom = nom;
        this.prix = prix;
        this.description = description;
        this.garantie = garantie;
        this.nomCat = nomCat;
    } 
}
